ledger/store/src/helpers/memory/internal/map.rs

```diff
@@ -242,6 +242,21 @@ impl<
         core::iter::Map<indexmap::map::IntoIter<K, Option<V>>, fn((K, Option<V>)) -> (Cow<'a, K>, Option<Cow<'a, V>>)>;
     type Values = core::iter::Map<btree_map::IntoValues<Vec<u8>, V>, fn(V) -> Cow<'a, V>>;
 
+    ///
+    /// Returns the number of confirmed entries in the map.
+    ///
+    fn len_confirmed(&self) -> usize {
+        self.map.read().len()
+    }
+
+    ///
+    /// Returns the number of pending entries in the map.
+    ///
+    fn len_pending(&self) -> usize {
+        let filtered_atomic_batch: HashSet<_> = self.atomic_batch.lock().clone().into_iter().map(|(k, _)| k).collect();
```

Comment:
I believe this is insufficient because it doesn't account for the case of removed keys (where the value in the `atomic_batch` is None).

Reply:
I dont think `len_pending` makes semantic sense.

Let's just remove all usage of `len_pending` for now.